File: app/src/main/java/com/njxm/smart/http/bean/BaseResponse.java
package com.njxm.smart.http.bean;

/**
 * @author huangxin
 * @date 2022/2/27
 */
public class BaseResponse<T> {
    private boolean isSuccess;

    private int code;

    private String error;

    private T data;

    public BaseResponse() {
    }

    public boolean isSuccess() {
        return isSuccess;
    }

    public void setSuccess(boolean success) {
        isSuccess = success;
    }

    public int getCode() {
        return code;
    }

    public void setCode(int code) {
        this.code = code;
    }

    public String getError() {
        return error;
    }

    public void setError(String error) {
        this.error = error;
    }

    public T getData() {
        return data;
    }

    public void setData(T data) {
        this.data = data;
    }
}

File: app/src/main/java/com/njxm/smart/network/base/BaseResponse.kt
package com.njxm.smart.network.base

/**
 *
 * @author huangxin
 * @date 2021/9/12
 */
class BaseResponse<T> {
    private var code: Int = 0;
    private var message: String = ""
    private val value: T? = null
}